api/turing/api/base_controller.go

```diff
@@ -85,3 +85,11 @@ func (c BaseController) getRouterVersionFromRequestVars(
 	}
 	return routerVersion, nil
 }
+
+func (c BaseController) getDeployFlagFromRequestVars(vars RequestVars) (deployFlag bool, error *Response) {
```

Comment:
This is weird. This deploy flag, unlike project/router, is only meaningful in the context of the `reouters_api` controller. Also, I'm not sure if it makes sense to define a separate function just for this, because this parameter is only extracted in a single controller's method (moving it into a function makes sense for the re-usability)